src/pool/pool_disjoint.c

```diff
@@ -381,10 +382,20 @@ static slab_list_item_t *bucket_get_avail_slab(bucket_t *bucket,
         // Allocation from existing slab is treated as from pool for statistics.
         *from_pool = true;
         if (slab->num_chunks_allocated == 0) {
+            assert(bucket->chunked_slabs_in_pool > 0);
+#ifndef NDEBUG
+            uint64_t total_size_check;
+            utils_atomic_load_acquire_u64(&bucket->shared_limits->total_size,
+                                          &total_size_check);
+            assert(total_size_check >= bucket_slab_alloc_size(bucket));
+#endif
             // If this was an empty slab, it was in the pool.
             // Now it is no longer in the pool, so update count.
             --bucket->chunked_slabs_in_pool;
-            bucket_decrement_pool(bucket);
+            ALIGNED_8 uint64_t size_to_add = bucket_slab_alloc_size(bucket);
```

Comment:
I think you put `ALIGNED_8` in many places where it is not needed (like here).

I think you can add debug assert to the `utils_*` functions for the atomic operations in the `utils_concurrency_posix.c` to make sure the variables are aligned. But I believe in almost all cases in this PR the `ALIGNED_8` is not needed.

Reply:
ok I removed all of them